### PR TITLE
load only apache config, fixes #168, fixes #169

### LINKED ISSUE #168: parsing cups printers.conf? wat?
I installed as described in the README on Ubuntu 14.04 and after starting letsencrypt from the venv, it shows the disclaimer and after I agree, it shows:

   There has been an error in parsing the file: /etc/cups/printers.conf  
   Parse error: mismatched in square lens, expecting "DefaultPrinter"   
   but got "Printer"                                                  
   No domain names were found in your installation                        
   Either specify which names you would like letsencrypt to validate    
   or add server names to your virtual hosts                         

Why is it parsing my cups printers.conf???

(the rest is ok, I have no hostnames, apache2 was freshly installed on this machine)

### LINKED ISSUE #169: running as root and reading about EVERY file in /etc
I did an strace on a letsencrypt run (see also #168) and found that it opens and reads about every file that lives in /etc (+ some more dotfiles).

Including, but not limited to, a lot of files that contain plaintext passwords, sudo configuration and other sensitive information that letsencrypt does definitely not need for operation. I expected it to look at apache/webserver stuff, but it also read ALL my wireless passwords (for example).

What would you think of a tool behaving like this?

Update: just to clarify: it is not the letsencrypt code that reads all the files, but the default behaviour of the augeas config parsing/editing library that is used by letsencrypt.

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
before it read almost ALL the files in /etc, including lots of files that contain passwords and other sensitive information.

please check my changes, I am new to augeas and also I did the development behind a firewall, so I could only partially test it. i have verified it does not access any unrelated files in /etc any more and it finds the virtualhosts definitions. could not get a cert, though due to firewall.

### CODE REVIEW

File: letsencrypt/client/augeas_configurator.py [hunk 1]
```diff
@@ -36,10 +36,16 @@ def __init__(self, direc=None):
                      "progress": CONFIG.IN_PROGRESS_DIR}
 
         self.direc = direc
-        # TODO: this instantiation can be optimized to only load
-        #       relevant files - I believe -> NO_MODL_AUTOLOAD
-        # Set Augeas flags to save backup
-        self.aug = augeas.Augeas(flags=augeas.Augeas.NONE)
+        # TODO improve this, if possible:
+        # incl should not be given here in this form as the files to look for are defined in httpd.aug
```
Comment (kuba) [inline letsencrypt/client/augeas_configurator.py]: that's way more than maximum 80 characters

File: letsencrypt/client/augeas_configurator.py [hunk 2]
```diff
@@ -399,7 +405,7 @@ def _remove_contained_files(self, file_list):
                     else:
                         logging.warn(
                             "File: %s - Could not be found to be deleted\n"
-                            "Program was probably shut down unexpectedly, ")
+                            "Program was probably shut down unexpectedly, " % path)
```
Comment (kuba) [inline letsencrypt/client/augeas_configurator.py]: please use lazy logging: `, path` instead of `% path`

**Comment by kuba** [comment]
@ThomasWaldmann please have a look at https://github.com/letsencrypt/lets-encrypt-preview#coding-style, thanks! :)

File: letsencrypt/client/augeas_configurator.py (hunk 1 above)
Comment (ThomasWaldmann) [inline letsencrypt/client/augeas_configurator.py]: sorry, i am used to follow pep8, but ignore that vt80 thing.

**Comment by jdkasten** [comment]
Fixed in #170.  Now only the files that are directly included via the configuration are parsed by Augeas. (The server root config + all files referenced recursively through Include directives)

**Pull request closed without merging**